projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts

```diff
@@ -508,6 +518,63 @@ export class GraphComponent extends BaseChartComponent implements OnInit, OnChan
     this.cd.markForCheck();
   }
 
+  getMinimapTransform(): string {
+    switch (this.miniMapPosition) {
+      case MiniMapPosition.UpperLeft: {
+        return '';
+      }
+      case MiniMapPosition.UpperRight: {
+        return 'translate(' + (this.dims.width - this.graphDims.width / this.minimapScaleCoefficient) + ',' + 0 + ')';
+      }
+      default: {
+        return '';
+      }
+    }
+  }
+
+  updateGraphDims() {
+    let minX = +Infinity;
+    let maxX = -Infinity;
+    let minY = +Infinity;
+    let maxY = -Infinity;
+
+    for (let i = 0; i < this.graph.nodes.length; i++) {
+      const node = this.graph.nodes[i];
+      minX = node.position.x < minX ? node.position.x : minX;
+      minY = node.position.y < minY ? node.position.y : minY;
+      maxX = node.position.x + node.dimension.width > maxX ? node.position.x + node.dimension.width : maxX;
+      maxY = node.position.y + node.dimension.height > maxY ? node.position.y + node.dimension.height : maxY;
+    }
+    minX -= 100;
+    minY -= 100;
+    maxX += 100;
+    maxY += 100;
+    this.graphDims.width = maxX - minX;
+    this.graphDims.height = maxY - minY;
+    this.minimapOffsetX = minX;
+    this.minimapOffsetY = minY;
+  }
+
+  @throttleable(500)
```

Comment:
interesting, what is throttleable?

Reply:
It gets recalculated on every mouse move event while you're dragging a node. So I'm throttling it to prevent calling it a huge number of times in succession.